src/machine.rs

```diff
@@ -498,7 +500,17 @@ impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
         let layouts =
             PrimitiveLayouts::new(layout_cx).expect("Couldn't get layouts of primitive types");
         let profiler = config.measureme_out.as_ref().map(|out| {
-            measureme::Profiler::new(out).expect("Couldn't create `measureme` profiler")
+            let crate_name = layout_cx
+                .tcx
+                .sess
+                .opts
+                .crate_name
+                .clone()
+                .unwrap_or_else(|| "unknown-crate".to_string());
+            let pid = process::id();
+            let filename = format!("{crate_name}-{pid:07}");
```

Comment:
Why are you padding the PID?

(Other tools using the PID in the filename, such as callgrind, AFAIK do not pad it.)

Reply:
Then please also copy the comment from rustc that explains this. :)

Reply:
The justification in rustc does not apply to Miri. It is explained in this hackmd: https://hackmd.io/sH315lO2RuicY-SEt7ynGA?view#ASLR-it%E2%80%99s-free-entropy The comment in the source assumes you know the context in the hackmd, which I didn't know until I asked other people.

We aren't doing high-precision instruction counting with ASLR off. I copied the implementation from rustc just for consistency, nothing else.